### PR TITLE
GLSP-1009: Provide web-extension example

### PR DESCRIPTION
**Pull request by tortmayr**
- Refactor codebase to provide entry points for both "normal" node VSCode extensions and web VSCode extensions. (via "@eclipse-glsp/vscode-integration/browser")
- Add additonal webview-extension example that covers two usecases
  - Communication with a GLSPServer that is running in the same extension context
  - Communication with a GLSPServer that is running externally via websocket
- Add launch configuration for starting the web extension

### CODE REVIEW

File: example/workflow/web-extension/src/server/main.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import 'reflect-metadata';
+
+import { WorkflowDiagramModule, WorkflowServerModule } from '@eclipse-glsp-examples/workflow-server/browser';
+import { LogLevel, createAppModule } from '@eclipse-glsp/server/browser';
+import { ContainerModule } from 'inversify';
+import { GModelVSCodeStorage } from './gmodel-vscode-model-storage';
+
+export function createServerModules(): ContainerModule[] {
```
Comment (martin-fleck-at) [inline example/workflow/web-extension/src/server/main.ts:23]: Could you please explain why this function is in a separate file? In the regular extension case, it is not and I don't see the benefit of it. Especially the name `main.ts` seems to imply that this is somehow compiled separately but I couldn't find anything related to it.

File: example/workflow/web-extension/src/workflow-extension.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const webSocketAddress = 'ws://localhost:8081/workflow';
+    const useWebSocket = await isAvailable(webSocketAddress);
+
+    const workflowServer = useWebSocket
+        ? new WebSocketGlspVscodeServer({
+              clientId: 'glsp.workflow',
+              clientName: 'workflow',
+              connectionOptions: { webSocketAddress }
+          })
+        : new NodeGlspVscodeServer({
```
Comment (martin-fleck-at) [inline example/workflow/web-extension/src/workflow-extension.ts:41]: The name `NodeGlspVscodeServer` is a bit confusing.

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Great work, thank you very much Tobias! It works well, I just have a few questions about the structure. I also noticed that we skipped the Elk layouting here, so if that causes issues we should open a follow-up ticket for it.

**Comment by tortmayr** [comment]
Thanks Martin, I have addressed your feedback.
The `NodeGlspVscodeServer` is now only used in the node context in web/brower its called `BrowserGlspVscodeServer`.
In addition, I have restructured the web extension server code into a single file `workflow-server.ts`

**Pull request merged**